### PR TITLE
98 rename @timestamp column

### PR DESCRIPTION
**Pull request by njuguna-n**
# Description

Rename the @timestamp column.

This PR depends on [this](https://github.com/medic/cht-sync/pull/117) on in the cht-sync repo.

Closes https://github.com/medic/cht-pipeline/issues/98

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by njuguna-n** [comment]
@lorerod please review.

File: models/contacts/contact.sql [hunk 1]
```diff
@@ -3,8 +3,8 @@
     materialized = 'incremental',
     unique_key='uuid',
     indexes=[
-      {'columns': ['"uuid"'], 'type': 'hash'},
-      {'columns': ['"@timestamp"']},
+      {'columns': ['uuid'], 'type': 'hash'},
+      {'columns': ['savedTimestamp']},
```
Comment (lorerod) [inline models/contacts/contact.sql:7]:
```suggestion
      {'columns': ['saved_timestamp']},
```

File: models/contacts/contact.sql [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 SELECT
   _id as uuid,
-  "@timestamp",
+  savedTimestamp,
```
Comment (lorerod) [inline models/contacts/contact.sql:17]:
```suggestion
  saved_timestamp,
```

File: models/contacts/contact.sql [hunk 3]
```diff
@@ -27,5 +27,5 @@ SELECT
 FROM {{ env_var('POSTGRES_SCHEMA') }}.{{ env_var('POSTGRES_TABLE') }}
 WHERE doc->>'type' IN ('contact', 'clinic', 'district_hospital', 'health_center', 'person')
 {% if is_incremental() %}
-  AND "@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/contacts/contact.sql:30]:
```suggestion
  AND saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: models/contacts/patient.sql [hunk 4]
```diff
@@ -3,20 +3,20 @@
     materialized = 'incremental',
     unique_key='uuid',
     indexes=[
-      {'columns': ['"uuid"'], 'type': 'hash'},
-      {'columns': ['"@timestamp"'], 'type': 'btree'},
+      {'columns': ['uuid'], 'type': 'hash'},
+      {'columns': ['savedTimestamp'], 'type': 'btree'},
```
Comment (lorerod) [inline models/contacts/patient.sql:7]:
```suggestion
      {'columns': ['saved_timestamp'], 'type': 'btree'},
```

File: models/contacts/patient.sql [hunk 5]
```diff
@@ -3,20 +3,20 @@
...
       {'columns': ['patient_id'], 'type': 'hash'},
     ]
   )
 }}
 
 SELECT
   uuid,
-  person."@timestamp",
+  person.savedTimestamp,
```
Comment (lorerod) [inline models/contacts/patient.sql:15]:
```suggestion
  person.saved_timestamp,
```

File: models/contacts/patient.sql [hunk 6]
```diff
@@ -3,20 +3,20 @@
...
   couchdb.doc->>'patient_id' as patient_id
 FROM {{ ref('person') }} person
 INNER JOIN {{ env_var('POSTGRES_SCHEMA') }}.{{ env_var('POSTGRES_TABLE') }} couchdb ON couchdb._id = uuid
 WHERE couchdb.doc->>'patient_id' IS NOT NULL
 {% if is_incremental() %}
-  AND person."@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND person.savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/contacts/patient.sql:21]:
```suggestion
  AND person.saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: models/contacts/person.sql [hunk 7]
```diff
@@ -4,19 +4,19 @@
     unique_key='uuid',
     indexes=[
       {'columns': ['uuid'], 'type': 'hash'},
-      {'columns': ['"@timestamp"']},
+      {'columns': ['savedTimestamp']},
```
Comment (lorerod) [inline models/contacts/person.sql:7]:
```suggestion
      {'columns': ['saved_timestamp']},
```

File: models/contacts/person.sql [hunk 8]
```diff
@@ -4,19 +4,19 @@
...
     ]
   )
 }}
 
 SELECT
   contact.uuid,
-  contact."@timestamp",
+  contact.savedTimestamp,
```
Comment (lorerod) [inline models/contacts/person.sql:14]:
```suggestion
  contact.saved_timestamp,
```

File: models/contacts/person.sql [hunk 9]
```diff
@@ -4,19 +4,19 @@
...
   couchdb.doc->>'date_of_birth' as date_of_birth,
   couchdb.doc->>'sex' as sex
 FROM {{ ref("contact") }} contact
 INNER JOIN {{ env_var('POSTGRES_SCHEMA') }}.{{ env_var('POSTGRES_TABLE') }} couchdb ON couchdb._id = uuid
 WHERE contact.contact_type = 'person'
 {% if is_incremental() %}
-  AND contact."@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND contact.savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/contacts/person.sql:21]:
```suggestion
  AND contact.saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: models/contacts/place.sql [hunk 10]
```diff
@@ -3,16 +3,16 @@
     materialized = 'incremental',
     unique_key='uuid',
     indexes=[
-      {'columns': ['"uuid"'], 'type': 'hash'},
-      {'columns': ['"@timestamp"']},
+      {'columns': ['uuid'], 'type': 'hash'},
+      {'columns': ['savedTimestamp']},
```
Comment (lorerod) [inline models/contacts/place.sql:7]:
```suggestion
      {'columns': ['saved_timestamp']},
```

File: models/contacts/place.sql [hunk 11]
```diff
@@ -3,16 +3,16 @@
...
       {'columns': ['place_id']},
     ]
   )
 }}
 
 SELECT
   uuid,
-  contact."@timestamp",
+  contact.savedTimestamp,
```
Comment (lorerod) [inline models/contacts/place.sql:15]:
```suggestion
  contact.saved_timestamp,
```

File: models/contacts/place.sql [hunk 12]
```diff
@@ -22,5 +22,5 @@ WHERE
     (contact.contact_type <> 'person')
   )
 {% if is_incremental() %}
-  AND contact."@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND contact.savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/contacts/place.sql:25]:
```suggestion
  AND contact.saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: models/forms/data_record.sql [hunk 13]
```diff
@@ -3,20 +3,20 @@
...
     indexes=[
-      {'columns': ['"uuid"'], 'type': 'hash'},
-      {'columns': ['"@timestamp"']},
-      {'columns': ['"reported"']},
-      {'columns': ['"from_phone"']},
-      {'columns': ['"form"']},
-      {'columns': ['"patient_id"']},
-      {'columns': ['"contact_uuid"']},
+      {'columns': ['uuid'], 'type': 'hash'},
+      {'columns': ['savedTimestamp']},
```
Comment (lorerod) [inline models/forms/data_record.sql:7]:
```suggestion
      {'columns': ['saved_timestamp']},
```

File: models/forms/data_record.sql [hunk 14]
```diff
@@ -3,20 +3,20 @@
...
+      {'columns': ['patient_id']},
+      {'columns': ['contact_uuid']},
     ]
   )
 }}
 
 SELECT
   _id as uuid,
-  "@timestamp",
+  savedTimestamp,
```
Comment (lorerod) [inline models/forms/data_record.sql:19]:
```suggestion
  saved_timestamp,
```

File: models/forms/data_record.sql [hunk 15]
```diff
@@ -39,5 +39,5 @@ FROM {{ env_var('POSTGRES_SCHEMA') }}.{{ env_var('POSTGRES_TABLE') }}
 WHERE
   doc->>'type' = 'data_record'
 {% if is_incremental() %}
-  AND "@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/forms/data_record.sql:42]:
```suggestion
  AND saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: models/users/user.sql [hunk 16]
```diff
@@ -4,14 +4,14 @@
     unique_key='user_id',
     indexes=[
       {'columns': ['user_id'], 'type': 'hash'},
-      {'columns': ['"@timestamp"']},
+      {'columns': ['savedTimestamp']},
```
Comment (lorerod) [inline models/users/user.sql:7]:
```suggestion
      {'columns': ['saved_timestamp']},
```

File: models/users/user.sql [hunk 17]
```diff
@@ -4,14 +4,14 @@
...
     ]
   )
 }}
 
 SELECT
   _id as user_id,
-  "@timestamp",
+  savedTimestamp,
```
Comment (lorerod) [inline models/users/user.sql:14]:
```suggestion
  saved_timestamp,
```

File: models/users/user.sql [hunk 18]
```diff
@@ -21,5 +21,5 @@ SELECT
 FROM {{ env_var('POSTGRES_SCHEMA') }}.{{ env_var('POSTGRES_TABLE') }}
 WHERE doc->>'type' = 'user-settings'
 {% if is_incremental() %}
-  AND "@timestamp" >= {{ max_existing_timestamp('"@timestamp"') }}
+  AND savedTimestamp >= {{ max_existing_timestamp('savedTimestamp') }}
```
Comment (lorerod) [inline models/users/user.sql:24]:
```suggestion
  AND saved_timestamp >= {{ max_existing_timestamp('saved_timestamp') }}
```

File: tests/setup.sh [hunk 19]
```diff
@@ -29,7 +29,7 @@ psql -h localhost -p 5432 -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbnam
     CREATE SCHEMA IF NOT EXISTS $POSTGRES_SCHEMA AUTHORIZATION $DBT_POSTGRES_USER;
     CREATE TABLE IF NOT EXISTS $POSTGRES_SCHEMA.$POSTGRES_TABLE(
       "@version" text NULL,
-      "@timestamp" timestamp NULL,
+      savedTimestamp timestamp NULL,
```
Comment (lorerod) [inline tests/setup.sh:32]:
```suggestion
      saved_timestamp timestamp NULL,
```

**Review by lorerod — CHANGES_REQUESTED** [review]
@njuguna-n, thank you for these changes. I suggest using lowercase with an underscore naming convention for Postgresql table column names. Let me know what you think.

**Review by lorerod — APPROVED** [review]

**Pull request merged**